### PR TITLE
[OpenAPI Learn Page ]Update openAPI learn page

### PR DESCRIPTION
**Pull request by lnash94**
Part of: https://github.com/ballerina-platform/ballerina-library/issues/6871

## Checklist

- [ ] **Page addition**
  - [ ] Add `permalink` to pages.

- [ ] **Page removal**
  - [ ] Remove entry from corresponding left nav YAML file.
  - [ ] Add `redirect_from` on the alternative page.
  - [ ] If no alternative page, add redirection on the `redirections.js ` file.

- [ ] **Page rename**
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

- [ ] **Page restrcuture**
  - [ ] Add `permalink` to pages.
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

### CODE REVIEW

File: swan-lake/integration-tools/openapi-tool.md [hunk 1]
```diff
@@ -312,18 +783,18 @@ options.license = "./license.txt"
...
-| `options.statusCodeBinding`| This option can be used in the client generation to generate the client methods with status code response binding.                                                                                                                                                                                                                                                                                                                                                                                           |  Optional         |
+| Command option             | Description                                                                                                                                                                                                                                                                                                                                                                                                 | Mandatory/Optional |
```
Comment (TharmiganK) [inline swan-lake/integration-tools/openapi-tool.md:786]: It should be `Field name` right?

File: swan-lake/integration-tools/openapi-tool.md [hunk 2]
```diff
@@ -406,15 +884,22 @@ service /greet on new http:Listener(9090) {
...
-| `Version: string?`             | Adds the version of the `info` section in the generated OpenAPI contract.                                                                                                                                                                                                                                                                 | Optional          |
-| `Embed: string?`               | Turns off generating OpenAPI documentation for the service for introspection endpoint support when used with `false` in the annotation.                                                                                                                                                                                                   | Optional          |
+| Command option                 | Description                                                                                                                                                                                                                                                                                                                               | Mandatory/Optional|
```
Comment (TharmiganK) [inline swan-lake/integration-tools/openapi-tool.md:887]: Here also, it should be the `Field name` right?

File: swan-lake/integration-tools/openapi-tool.md [hunk 3]
```diff
@@ -406,15 +884,22 @@ service /greet on new http:Listener(9090) {
...
+|--------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------|
+| `contract: string?`            | Provides a path to the OpenAPI contract as a string and the OpenAPI file can either be a `.yaml` or `.json`.                                                                                                                                                                                                                              | Mandatory         |
```
[the remark this answers is not in the document]
Comment (lnash94) [reply swan-lake/integration-tools/openapi-tool.md:903]: Yeah we can remove the redundant info

File: swan-lake/integration-tools/openapi-tool.md [hunk 4]
```diff
@@ -5,7 +5,7 @@ description: Check out how the Ballerina OpenAPI tool makes it easy for you to s
 keywords: ballerina, programming language, openapi, open api, restful api
 permalink: /learn/openapi-tool/
 active: openapi-tool
-intro: OpenAPI Specification is a specification that creates a RESTFUL contract for APIs by detailing all of its resources and operations in a human and machine-readable format for easy development, discovery, and integration. Ballerina Swan Lake supports the OpenAPI Specification version 3.0.0 onwards.
+intro: OpenAPI contract is a specification that creates a RESTFUL contract for APIs by detailing all of its resources and operations in a human and machine-readable format for easy development, discovery, and integration. Ballerina Swan Lake supports the OpenAPI contract version 3.0.0 onwards.
```
Comment (MaryamZi) [inline swan-lake/integration-tools/openapi-tool.md:8]:
```suggestion
intro: The OpenAPI contract is a specification that creates a RESTful contract for APIs by detailing all of its resources and operations in a human and machine-readable format for easy development, discovery, and integration. Ballerina Swan Lake supports the OpenAPI contract version 3.0.0 onwards.
```

File: swan-lake/integration-tools/openapi-tool.md [hunk 5]
```diff
@@ -53,34 +54,40 @@ The below command-line arguments can be used with the command.
...
+| `--client-methods`        | This option can be used in the client generation to select the client method type, which can be `resource` or `remote`. (The default option is `resource`).                                                                                                                                                                                                                                                                                                                                                                                           |  Optional         |
+| `--status-code-binding`   | This option can be used in the client generation to generate the client methods with status code response binding. | Optional |
+| `--mock`                  | This flag can be used to switch client generation to generate a mock client for the given OpenAPI contract.  | Optional |
```
Comment (MaryamZi) [inline swan-lake/integration-tools/openapi-tool.md:69]: Seems like option is used consistently with the others?

File: swan-lake/integration-tools/openapi-tool.md [hunk 6]
```diff
@@ -53,34 +54,40 @@ The below command-line arguments can be used with the command.
...
+|`--with-service-contract`  | This option can be used to generate the service contract type for the given OpenAPI contract. | Optional |
+| `--single-file`           | This option can be used to generate the Ballerina service or client with related types and utility functions in a single file. | Optional |
+| `--use-sanitized-oas`     | This is an experimental feature. This option enables service/client code generation by modifying the given OAS to follow the Ballerina language best practices. | Optional |
```
Comment (MaryamZi) [inline swan-lake/integration-tools/openapi-tool.md:72]:
```suggestion
| `--use-sanitized-oas`     | This is an experimental feature. This option enables service/client code generation by modifying the given OAS to follow Ballerina language best practices. | Optional |
```

File: swan-lake/integration-tools/openapi-tool.md [hunk 7]
```diff
@@ -90,17 +97,63 @@ The generated service can be used as a code template to start the service implem
...
+import ballerina/http;
+
+listener http:Listener ep0 = new (9090, config = {host: "localhost"});
+
+service /store on ep0 {
+    resource function get store() returns Inventory|http:Accepted {
+        // logic here
+    }
+
+    resource function post store(@http:Payload Inventory payload) {
```
Comment (MaryamZi) [inline swan-lake/integration-tools/openapi-tool.md:115]: Is the annotation still included?

File: swan-lake/integration-tools/openapi-tool.md [hunk 8]
```diff
@@ -90,17 +97,63 @@ The generated service can be used as a code template to start the service implem
...
+```ballerina
+import ballerina/http;
 
+@http:ServiceConfig {basePath: "/store"}
+type OASServiceType service object {
+    *http:ServiceContract;
+    resource function get store() returns Inventory|http:Accepted;
+    resource function post store(@http:Payload Inventory payload) returns http:Accepted|ErrorPayloadBadRequest;
+};
+```
```
Comment (MaryamZi) [inline swan-lake/integration-tools/openapi-tool.md:149]:
````suggestion
```

````

File: swan-lake/integration-tools/openapi-tool.md [hunk 9]
```diff
@@ -90,17 +97,63 @@ The generated service can be used as a code template to start the service implem
...
 
 ```
 The service generation process is complete. The following files were created.
--- hello_service.bal
+-- store_service.bal
 -- types.bal
 ```
+### Generate service contract object for given OpenAPI contract
+
+OpenAPI Specification (OAS) is a way to describe the service interface of REST APIs. Counterpart of this in Ballerina is service object type. This `--with-service-contract` is to enable code generation for Ballerina service contract object.
```
Comment (MaryamZi) [inline swan-lake/integration-tools/openapi-tool.md:130]:
```suggestion
OpenAPI Specification (OAS) is a way to describe the service interface of REST APIs. Counterpart of this in Ballerina is service object types. This `--with-service-contract` is to enable code generation for the Ballerina service contract object type.
```

File: swan-lake/integration-tools/openapi-tool.md (hunk 8 above)
Comment (MaryamZi) [reply swan-lake/integration-tools/openapi-tool.md:149]: Please check everywhere.

File: swan-lake/integration-tools/openapi-tool.md [hunk 10]
```diff
@@ -117,11 +170,22 @@ $ bal openapi -i hello.yaml --tags "pets", "list"
 
 Once you execute the command, only the operations related to the given tags get included in the generated service file.
 
+## Generate Ballerina service contract type from OpenAPI Contracts
+
+A service contract-first approach is introduced as a standout key feature using the OpenAPI tool and the HTTP package. By generating a service contract type directly from the OpenAPI contract, you can design services that are contract-compliant from the outset.
```
Comment (MaryamZi) [inline swan-lake/integration-tools/openapi-tool.md:175]: IMO, things like "A service contract-first approach is introduced as a standout key feature using the OpenAPI tool and the HTTP package." are for the release note, not for generic documentation.

File: swan-lake/integration-tools/openapi-tool.md [hunk 11]
```diff
@@ -117,11 +170,22 @@ $ bal openapi -i hello.yaml --tags "pets", "list"
...
 
+## Generate Ballerina service contract type from OpenAPI Contracts
+
+A service contract-first approach is introduced as a standout key feature using the OpenAPI tool and the HTTP package. By generating a service contract type directly from the OpenAPI contract, you can design services that are contract-compliant from the outset. 
+In addition, centralizing metadata within the service contract type results in a cleaner, more maintainable codebase.
+
+For example, 
+
+```
+$bal openapi -i hello.yaml --mode service --with-service-contract
```
Comment (MaryamZi) [inline swan-lake/integration-tools/openapi-tool.md:181]:
```suggestion
$ bal openapi -i hello.yaml --mode service --with-service-contract
```

**Review by TharmiganK — APPROVED** [review]

**Pull request merged**